@hydai Yes, the master branch build currently fails on my machine. I am not sure how this got through the CI. Here are the logs for the failing build:
```
 build $ cmake --version
cmake version 4.3.2

CMake suite maintained and supported by Kitware (kitware.com/cmake).
 build $ git log --oneline | head -5
b9ea2913 fix: storemanager::reset clears namedcomp (#4897)
7fb4cd62 refactor(executor): extract SIMD superinstruction primitives into header (#4802)
f1d20116 fix: fix lookup error while vm unregister module (#4905)
874c575e ci(dependabot): bump step-security/harden-runner from 2.19.3 to 2.19.4 (#4899)
8a248a6d feat: add tests for module dependency tree
 build $ cmake -DWASMEDGE_BUILD_TESTS=ON -DWASMEDGE_PLUGIN_IMAGE=ON .. && make -j 2
-- spdlog found
-- Downloading stb_image source
-- Downloading stb_image source -- done
-- Checking SIMDJSON from system
-- Downloading SIMDJSON source
-- Adding -Og to compile flag
-- Downloading SIMDJSON source -- done
-- Downloading the WASM spec test suite
-- Downloading the WASM spec test suite -- done
-- Copying test suite to /home/pranjal/Projects/WasmEdge/build/test/spec/testSuites/wasm-1.0
-- Copying test suite to /home/pranjal/Projects/WasmEdge/build/test/spec/testSuites/wasm-1.0 -- done
-- Copying test suite to /home/pranjal/Projects/WasmEdge/build/test/spec/testSuites/wasm-2.0
-- Copying test suite to /home/pranjal/Projects/WasmEdge/build/test/spec/testSuites/wasm-2.0 -- done
-- Copying test suite to /home/pranjal/Projects/WasmEdge/build/test/spec/testSuites/wasm-3.0
-- Copying test suite to /home/pranjal/Projects/WasmEdge/build/test/spec/testSuites/wasm-3.0 -- done
-- Copying test suite to /home/pranjal/Projects/WasmEdge/build/test/spec/testSuites/wasm-3.0-bulk-memory
-- Copying test suite to /home/pranjal/Projects/WasmEdge/build/test/spec/testSuites/wasm-3.0-bulk-memory -- done
-- Copying test suite to /home/pranjal/Projects/WasmEdge/build/test/spec/testSuites/wasm-3.0-exceptions
-- Copying test suite to /home/pranjal/Projects/WasmEdge/build/test/spec/testSuites/wasm-3.0-exceptions -- done
-- Copying test suite to /home/pranjal/Projects/WasmEdge/build/test/spec/testSuites/wasm-3.0-gc
-- Copying test suite to /home/pranjal/Projects/WasmEdge/build/test/spec/testSuites/wasm-3.0-gc -- done
-- Copying test suite to /home/pranjal/Projects/WasmEdge/build/test/spec/testSuites/wasm-3.0-memory64
-- Copying test suite to /home/pranjal/Projects/WasmEdge/build/test/spec/testSuites/wasm-3.0-memory64 -- done
-- Copying test suite to /home/pranjal/Projects/WasmEdge/build/test/spec/testSuites/wasm-3.0-multi-memory
-- Copying test suite to /home/pranjal/Projects/WasmEdge/build/test/spec/testSuites/wasm-3.0-multi-memory -- done
-- Copying test suite to /home/pranjal/Projects/WasmEdge/build/test/spec/testSuites/wasm-3.0-relaxed-simd
-- Copying test suite to /home/pranjal/Projects/WasmEdge/build/test/spec/testSuites/wasm-3.0-relaxed-simd -- done
-- Copying test suite to /home/pranjal/Projects/WasmEdge/build/test/spec/testSuites/wasm-3.0-simd
-- Copying test suite to /home/pranjal/Projects/WasmEdge/build/test/spec/testSuites/wasm-3.0-simd -- done
-- Copying test suite to /home/pranjal/Projects/WasmEdge/build/test/spec/testSuites/threads
-- Copying test suite to /home/pranjal/Projects/WasmEdge/build/test/spec/testSuites/threads -- done
-- Copying test suite to /home/pranjal/Projects/WasmEdge/build/test/spec/testSuites/component-model
-- Copying test suite to /home/pranjal/Projects/WasmEdge/build/test/spec/testSuites/component-model -- done
-- Configuring done (3.3s)
-- Generating done (0.4s)
-- Build files have been written to: /home/pranjal/Projects/WasmEdge/build
[  4%] Built target utilBlake3
[  6%] Built target wasmedgeCommon
[  7%] Built target simdjson
[ 12%] Built target expectedTests
[ 12%] Built target spanTests
[ 13%] Built target wasmedgeErrinfoTests
[ 16%] Built target wasmedgeSystem
[ 17%] Built target wasmedgePluginWasiLogging
[ 18%] Built target wasmedgePO
[ 20%] Built target wasmedgeValidator
[ 23%] Built target wasmedgeHostModuleWasi
[ 35%] Built target wasmedgeExecutor
[ 35%] Built target wasmedgeTestSpec
[ 37%] Built target wasmedgeCommonTests
[ 39%] Built target wasiTests
[ 41%] Built target poTests
[ 42%] Built target wasmedgeAOT
[ 44%] Built target wasmedgeLoaderFileMgr
[ 46%] Built target wasmedgeLLVM
[ 47%] Built target wasmedgeAOTCacheTests
[ 48%] Built target wasmedgeAOTBlake3Tests
[ 48%] Built target wasmedgeLoaderFileMgrTests
[ 49%] Built target wasmedgePlugin
[ 61%] Built target wasmedgeLoader
[ 62%] Built target wasmedgeVM
[ 65%] Built target wasmedgeLoaderASTTests
[ 67%] Built target wasmedgeExecutorCoreTests
[ 70%] Built target wasmedgeLoaderSerializerTests
[ 72%] Built target wasmedgeExecutorRegressionTests
[ 72%] Built target wasmedgeValidatorRegressionTests
[ 72%] Built target wasmedgeComponentFuzzTest
[ 74%] Built target wasmedgeComponentRegressionTests
[ 75%] Built target wasmedgeThreadTests
[ 75%] Built target wasiSocketTests
[ 77%] Built target wasmedgeMemInstanceTests
[ 77%] Built target wasmedgeHostMockTests
[ 78%] Built target wasmedgeLLVMCoreTests
[ 82%] Built target wasmedgeDriver
[ 83%] Built target wasmedgeLazyJITTests
[ 83%] Built target wasmedgeMixcallTests
[ 84%] Built target wasmedgeDriverToolTests
[ 84%] Built target wasmedgeCAPI
[ 85%] Built target wasmedge_shared
[ 86%] Built target wasmedgec
[ 88%] Built target wasmedgePluginWasmEdgeImage
[ 89%] Built target wasmedge
[ 90%] Built target wasmedgeAPITestHelpers
[ 91%] Built target wasmedgeExternrefTests
[ 92%] Built target wasiLoggingTests
[ 92%] Linking CXX executable wasmedgeImageTests
[ 93%] Built target wasmedgePluginTestModuleC
[ 94%] Built target wasmedgePluginTestModuleCPP
[ 94%] Built target wasmedgePluginUnittestsC
[ 95%] Built target wasmedgePluginUnittestsCPP
[ 96%] Built target wasmedgeAPIUnitTests
[ 97%] Built target wasmedgeAPIVMCoreTests
[ 98%] Built target wasmedgeAPIStepsCoreTests
[ 99%] Built target wasmedgeAPIAOTCoreTests
[ 99%] Built target wasmedgeAPIAOTNestedVMTests
/usr/bin/ld: /tmp/ccSl9l2V.ltrans0.ltrans.o: in function `WasmEdge::Runtime::HostFunction<WasmEdge::Host::WasmEdgeImage::LoadJPG>::run(WasmEdge::Runtime::CallingFrame const&, cxx20::span<WasmEdge::Variant<unsigned int, int, unsigned long, long, float, double, unsigned __int128, __int128, unsigned long __vector(2), long __vector(2), unsigned int __vector(4), int __vector(4), unsigned short __vector(8), short __vector(8), unsigned char __vector(16), signed char __vector(16), float __vector(4), double __vector(2), WasmEdge::RefVariant> const, 18446744073709551615ul>, cxx20::span<WasmEdge::Variant<unsigned int, int, unsigned long, long, float, double, unsigned __int128, __int128, unsigned long __vector(2), long __vector(2), unsigned int __vector(4), int __vector(4), unsigned short __vector(8), short __vector(8), unsigned char __vector(16), signed char __vector(16), float __vector(4), double __vector(2), WasmEdge::RefVariant>, 18446744073709551615ul>)':
/usr/include/c++/16.1.1/bits/invoke.h:69:(.text+0x6f): undefined reference to `WasmEdge::Host::WasmEdgeImage::LoadJPG::body(WasmEdge::Runtime::CallingFrame const&, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int)'
/usr/bin/ld: /tmp/ccSl9l2V.ltrans0.ltrans.o: in function `WasmEdge::Runtime::HostFunction<WasmEdge::Host::WasmEdgeImage::LoadPNG>::run(WasmEdge::Runtime::CallingFrame const&, cxx20::span<WasmEdge::Variant<unsigned int, int, unsigned long, long, float, double, unsigned __int128, __int128, unsigned long __vector(2), long __vector(2), unsigned int __vector(4), int __vector(4), unsigned short __vector(8), short __vector(8), unsigned char __vector(16), signed char __vector(16), float __vector(4), double __vector(2), WasmEdge::RefVariant> const, 18446744073709551615ul>, cxx20::span<WasmEdge::Variant<unsigned int, int, unsigned long, long, float, double, unsigned __int128, __int128, unsigned long __vector(2), long __vector(2), unsigned int __vector(4), int __vector(4), unsigned short __vector(8), short __vector(8), unsigned char __vector(16), signed char __vector(16), float __vector(4), double __vector(2), WasmEdge::RefVariant>, 18446744073709551615ul>)':
/usr/include/c++/16.1.1/bits/invoke.h:69:(.text+0x12f): undefined reference to `WasmEdge::Host::WasmEdgeImage::LoadPNG::body(WasmEdge::Runtime::CallingFrame const&, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int)'
/usr/bin/ld: /tmp/ccSl9l2V.ltrans0.ltrans.o: in function `WasmEdge::Runtime::HostFunction<WasmEdge::Host::WasmEdgeImage::LoadImage>::run(WasmEdge::Runtime::CallingFrame const&, cxx20::span<WasmEdge::Variant<unsigned int, int, unsigned long, long, float, double, unsigned __int128, __int128, unsigned long __vector(2), long __vector(2), unsigned int __vector(4), int __vector(4), unsigned short __vector(8), short __vector(8), unsigned char __vector(16), signed char __vector(16), float __vector(4), double __vector(2), WasmEdge::RefVariant> const, 18446744073709551615ul>, cxx20::span<WasmEdge::Variant<unsigned int, int, unsigned long, long, float, double, unsigned __int128, __int128, unsigned long __vector(2), long __vector(2), unsigned int __vector(4), int __vector(4), unsigned short __vector(8), short __vector(8), unsigned char __vector(16), signed char __vector(16), float __vector(4), double __vector(2), WasmEdge::RefVariant>, 18446744073709551615ul>)':
/usr/include/c++/16.1.1/bits/invoke.h:69:(.text+0x1ef): undefined reference to `WasmEdge::Host::WasmEdgeImage::LoadImage::body(WasmEdge::Runtime::CallingFrame const&, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int)'
/usr/bin/ld: /tmp/ccSl9l2V.ltrans0.ltrans.o: in function `WasmEdgeImageTest_LoadJPG_Test::TestBody()':
/usr/include/c++/16.1.1/bits/invoke.h:69:(.text+0x5a97): undefined reference to `WasmEdge::Host::WasmEdgeImage::LoadJPG::body(WasmEdge::Runtime::CallingFrame const&, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int)'
/usr/bin/ld: /usr/include/c++/16.1.1/bits/invoke.h:69:(.text+0x5ddd): undefined reference to `WasmEdge::Host::WasmEdgeImage::LoadJPG::body(WasmEdge::Runtime::CallingFrame const&, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int)'
/usr/bin/ld: /usr/include/c++/16.1.1/bits/invoke.h:69:(.text+0x6123): undefined reference to `WasmEdge::Host::WasmEdgeImage::LoadJPG::body(WasmEdge::Runtime::CallingFrame const&, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int)'
/usr/bin/ld: /usr/include/c++/16.1.1/bits/invoke.h:69:(.text+0x6358): undefined reference to `WasmEdge::Host::WasmEdgeImage::LoadJPG::body(WasmEdge::Runtime::CallingFrame const&, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int)'
/usr/bin/ld: /tmp/ccSl9l2V.ltrans0.ltrans.o: in function `WasmEdgeImageTest_LoadPNG_Test::TestBody()':
/usr/include/c++/16.1.1/bits/invoke.h:69:(.text+0x82f7): undefined reference to `WasmEdge::Host::WasmEdgeImage::LoadPNG::body(WasmEdge::Runtime::CallingFrame const&, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int)'
/usr/bin/ld: /usr/include/c++/16.1.1/bits/invoke.h:69:(.text+0x863d): undefined reference to `WasmEdge::Host::WasmEdgeImage::LoadPNG::body(WasmEdge::Runtime::CallingFrame const&, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int)'
/usr/bin/ld: /usr/include/c++/16.1.1/bits/invoke.h:69:(.text+0x8983): undefined reference to `WasmEdge::Host::WasmEdgeImage::LoadPNG::body(WasmEdge::Runtime::CallingFrame const&, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int)'
/usr/bin/ld: /usr/include/c++/16.1.1/bits/invoke.h:69:(.text+0x8bb8): undefined reference to `WasmEdge::Host::WasmEdgeImage::LoadPNG::body(WasmEdge::Runtime::CallingFrame const&, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int)'
/usr/bin/ld: /tmp/ccSl9l2V.ltrans0.ltrans.o: in function `WasmEdgeImageTest_LoadImage_Test::TestBody()':
/usr/include/c++/16.1.1/bits/invoke.h:69:(.text+0xab47): undefined reference to `WasmEdge::Host::WasmEdgeImage::LoadImage::body(WasmEdge::Runtime::CallingFrame const&, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int)'
/usr/bin/ld: /usr/include/c++/16.1.1/bits/invoke.h:69:(.text+0xae7d): undefined reference to `WasmEdge::Host::WasmEdgeImage::LoadImage::body(WasmEdge::Runtime::CallingFrame const&, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int, unsigned int)'
collect2: error: ld returned 1 exit status
make[2]: *** [test/plugins/wasmedge_image/CMakeFiles/wasmedgeImageTests.dir/build.make:124: test/plugins/wasmedge_image/wasmedgeImageTests] Error 1
make[1]: *** [CMakeFiles/Makefile2:4229: test/plugins/wasmedge_image/CMakeFiles/wasmedgeImageTests.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
```

--- a/test/plugins/wasmedge_image/wasmedge_image.cpp
+++ b/test/plugins/wasmedge_image/wasmedge_image.cpp
@@ -2,8 +2,8 @@
 // SPDX-FileCopyrightText: 2019-2024 Second State INC
 
 #include "common/defines.h"
-#include "image_func.h"
 #include "image_module.h"
+#include "runtime/callingframe.h"
 #include "runtime/instance/module.h"
 
 #include <algorithm>
@@ -179,8 +179,7 @@ TEST(WasmEdgeImageTest, LoadJPG) {
   auto *FuncInst = ImgMod->findFuncExports("load_jpg");
   EXPECT_NE(FuncInst, nullptr);
   EXPECT_TRUE(FuncInst->isHostFunction());
-  auto &HostFuncInst = dynamic_cast<WasmEdge::Host::WasmEdgeImage::LoadJPG &>(
-      FuncInst->getHostFunc());
+  auto &HostFuncInst = FuncInst->getHostFunc();
 
   // Test: Load JPG and resize into 50x60 RGB u8 format.
   // Clear the memory[0, 32768].
@@ -320,8 +319,7 @@ TEST(WasmEdgeImageTest, LoadPNG) {
   auto *FuncInst = ImgMod->findFuncExports("load_png");
   EXPECT_NE(FuncInst, nullptr);
   EXPECT_TRUE(FuncInst->isHostFunction());
-  auto &HostFuncInst = dynamic_cast<WasmEdge::Host::WasmEdgeImage::LoadPNG &>(
-      FuncInst->getHostFunc());
+  auto &HostFuncInst = FuncInst->getHostFunc();
 
   // Test: Load PNG and resize into 50x60 RGB u8 format.
   // Clear the memory[0, 32768].
@@ -459,8 +457,7 @@ TEST(WasmEdgeImageTest, LoadImage) {
   auto *FuncInst = ImgMod->findFuncExports("load_image");
   EXPECT_NE(FuncInst, nullptr);
   EXPECT_TRUE(FuncInst->isHostFunction());
-  auto &HostFuncInst = dynamic_cast<WasmEdge::Host::WasmEdgeImage::LoadImage &>(
-      FuncInst->getHostFunc());
+  auto &HostFuncInst = FuncInst->getHostFunc();
 
   // Test: Load JPG and resize into 50x60 BGR u8 format.
   // Clear the memory[0, 32768].

